Neos.ContentRepository/Classes/Security/Authorization/Privilege/Node/TransientNodeCache.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Security\Authorization\Privilege\Node;
+
+use Neos\ContentRepository\Domain\Model\NodeInterface;
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * Simple cache for nodes that is intended to be used in `NodePrivilegeContext`.
+ *
+ * WARNING: This cache may contain nodes that should not be visible to the current user.
+ *          Only use this cache in `NodePrivilegeContext`.
+ *
+ * @Flow\Scope("singleton")
+ */
+class TransientNodeCache
+{
+    private $nodes = [];
+
+    /**
+     * @param string $nodeIdentifier
+     * @return bool
+     */
+    private function has(string $nodeIdentifier): bool
```

Comment:
I would remove this private methods. IMHO it is a lot of code and indirection to wrap some simpel methods.